Automattic/wp-calypso
# Simplify the QueryMediaStorage component and related actions and state

Refactors the `QueryMediaStorage` component to a functional component with hooks.

The component doesn't need to call the `isRequestingMediaStorage` selector, because requests are already deduplicated by the data layer, and there are never multiple requests ran in parallel.

As this was the only usage of `isRequestingMediaStorage`, we can remove the associated state and actions. We only need the `_REQUEST` and `_RECEIVE` actions, and only the `state.sites.mediaStorage.items` state. This removes a lot of code.

**How to test:**
Open Media for your site and verify you see the storage quota indicator saying you're using "X% of Y GB". After deleting or updating a media item, verify that the storage quota info is re-requested.
